fast64_internal/oot/actor/properties.py

```diff
@@ -110,22 +164,193 @@ def draw_props(
 class OOTActorProperty(PropertyGroup):
     actorID: EnumProperty(name="Actor", items=ootData.actorData.ootEnumActorID, default="ACTOR_PLAYER")
     actorIDCustom: StringProperty(name="Actor ID", default="ACTOR_PLAYER")
+
+    # used for actors with the id "Custom"
     actorParam: StringProperty(name="Actor Parameter", default="0x0000")
     rotOverride: BoolProperty(name="Override Rotation", default=False)
-    rotOverrideX: StringProperty(name="Rot X", default="0")
-    rotOverrideY: StringProperty(name="Rot Y", default="0")
-    rotOverrideZ: StringProperty(name="Rot Z", default="0")
+    rotOverrideX: StringProperty(name="Rot X", default="0x0000")
+    rotOverrideY: StringProperty(name="Rot Y", default="0x0000")
+    rotOverrideZ: StringProperty(name="Rot Z", default="0x0000")
+
+    # non-custom actors
+    params: StringProperty(
+        name="Actor Parameter",
+        default="0x0000",
+        get=lambda self: self.getParamValue("Params"),
+        set=lambda self, value: self.setParamValue(value, "Params"),
+    )
+
+    rotX: StringProperty(
+        name="Rot X",
+        default="0",
+        get=lambda self: self.getParamValue("XRot"),
+        set=lambda self, value: self.setParamValue(value, "XRot"),
+    )
+    rotY: StringProperty(
+        name="Rot Y",
+        default="0",
+        get=lambda self: self.getParamValue("YRot"),
+        set=lambda self, value: self.setParamValue(value, "YRot"),
+    )
+    rotZ: StringProperty(
+        name="Rot Z",
+        default="0",
+        get=lambda self: self.getParamValue("ZRot"),
+        set=lambda self, value: self.setParamValue(value, "ZRot"),
+    )
+
+    # internal usage only
+    isRotXUsedByActor: BoolProperty(name="", default=False, get=lambda self: self.isRotationUsedByActor("XRot"))
+    isRotYUsedByActor: BoolProperty(name="", default=False, get=lambda self: self.isRotationUsedByActor("YRot"))
+    isRotZUsedByActor: BoolProperty(name="", default=False, get=lambda self: self.isRotationUsedByActor("ZRot"))
+
     headerSettings: PointerProperty(type=OOTActorHeaderProperty)
+    evalParams: BoolProperty(name="Eval Params", default=False)
 
     @staticmethod
     def upgrade_object(obj: Object):
         print(f"Processing '{obj.name}'...")
         upgradeActors(obj)
 
+    def isRotationUsedByActor(self, target: str):
+        actor = ootData.actorData.actorsByID[self.actorID]
+        curType = None
+        for param in actor.params:
+            if param.type == "Type":
+                objName = getObjName(actor.key, param.type, param.subType, param.index)
+                curType = getEvalParamsInt(getattr(self, objName))
+
+            if curType is not None and curType in param.tiedTypes or len(param.tiedTypes) == 0:
```

Comment:
Make this `elif`? Or maybe add a comment explaining that the first param is always of type `"Type"`, and that there is only one param like this. This code is a little confusing without this context.